store/tikv/rawkv.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tikv
+
+import (
+	"context"
+
+	"github.com/juju/errors"
+	"github.com/pingcap/kvproto/pkg/kvrpcpb"
+	"github.com/pingcap/pd/pd-client"
+)
+
+// RawKVClient is a client of TiKV server which is used as a key-value storage,
+// only GET/PUT/DELETE commands are supported.
+type RawKVClient struct {
+	clusterID   uint64
+	regionCache *RegionCache
+	rpcClient   Client
+}
+
+// NewRawKVClient creates a client with PD cluster addrs.
+func NewRawKVClient(pdAddrs []string) (*RawKVClient, error) {
+	pdCli, err := pd.NewClient(pdAddrs)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return &RawKVClient{
+		clusterID:   pdCli.GetClusterID(),
+		regionCache: NewRegionCache(pdCli),
+		rpcClient:   newRPCClient(),
+	}, nil
+}
+
+// ClusterID returns the TiKV cluster ID.
+func (c *RawKVClient) ClusterID() uint64 {
+	return c.clusterID
+}
+
+// Get queries value with the key. When the key does not exist, it returns
```

Comment:
This is not consisted with union store interface. For key not exist, it will return KeyNotExist error.
See: https://github.com/pingcap/tidb/blob/master/kv/union_store.go#L141